POST error response when using Entra as OIDC provider for Oauth

Trying out the TF provider (v.0.4.5) against an existing Lakekeeper instance, using MS Entra as OIDC provider:
```
terraform {
  required_providers {
    lakekeeper = {
      source  = "baptistegh/lakekeeper"
      version = "0.4.5"
    }
  }
}

provider "lakekeeper" {
  endpoint          = "https://<redacted>/catalog"
  auth_url          = "https://login.microsoftonline.com/09477d9c-<redacted>-fce12d6/oauth2/v2.0/token"
  client_id         = "<redacted>"
  client_secret     = "<redacted>"
  scopes            = ["email openid api://909ea26c-<redacted>-0814/.default"]
  initial_bootstrap = false
}

# Create the catalog
resource "lakekeeper_project" "hello" {
   name = "hello"
}
```

All the values Oauth match those of PyIceberg and Trino configurations where they work without issues, with the sensitive parts <redacted>. Using the TF provider though, I run into the following:


```
│ Error: Lakekeeper API error occurred
│ 
│   with lakekeeper_project.hello,
│   on main.tf line 20, in resource "lakekeeper_project" "hello":
│   20: resource "lakekeeper_project" "hello" {
│ 
│ Unable to create project hello, unexpected error response, Post
│ "https://login.microsoftonline.com/09477d9c-<redacted>-fce12d6/oauth2/v2.0/token": context canceled
```

with TF_LOG level to `debug`:

```
e.go:538 tf_req_id=9feb38c9-6e53-967f-814b-c438efd08412 tf_rpc=PlanResourceChange timestamp=2025-10-29T18:46:19.951-0400
lakekeeper_project.hello: Creating...
2025-10-29T18:46:19.951-0400 [INFO]  Starting apply for lakekeeper_project.hello
2025-10-29T18:46:19.951-0400 [DEBUG] lakekeeper_project.hello: applying the planned Create change
2025-10-29T18:46:19.952-0400 [ERROR] provider.terraform-provider-lakekeeper_v0.4.5: Response contains error diagnostic: tf_provider_addr=registry.terraform.io/baptistegh/lakekeeper tf_req_id=302846d2-d296-f794-f7eb-144a5ed67c25 @module=sdk.proto diagnostic_severity=ERROR 
tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov6/internal/diag/diagnostics.go:58 diagnostic_detail="Unable to create project hello, unexpected error response, Post \"https://login.microsoftonline.com/09477d9c-<redacted>-fce12d6/oauth2/v2.0/token\": context canceled" diagnostic_summary="Lakekeeper API error occurred" tf_proto_version=6.10 tf_resource_type=lakekeeper_project timestamp=2025-10-29T18:46:19.952-0400
2025-10-29T18:46:19.955-0400 [DEBUG] State storage *statemgr.Filesystem declined to persist a state snapshot
2025-10-29T18:46:19.955-0400 [ERROR] vertex "lakekeeper_project.hello" error: Lakekeeper API error occurred
```

--- a/internal/provider/api/config.go
+++ b/internal/provider/api/config.go
@@ -7,10 +7,12 @@ import (
 	"errors"
 	"net/http"
 	"os"
+	"time"
 
 	managementv1 "github.com/baptistegh/go-lakekeeper/pkg/apis/management/v1"
 	lakekeeper "github.com/baptistegh/go-lakekeeper/pkg/client"
 	"github.com/baptistegh/go-lakekeeper/pkg/core"
+	"golang.org/x/oauth2"
 	"golang.org/x/oauth2/clientcredentials"
 )
 
@@ -86,8 +88,13 @@ func (c *Config) NewLakekeeperClient(ctx context.Context) (*lakekeeper.Client, e
 		Scopes:       c.Scopes,
 	}
 
+	// configure context for OAuth client
+	oauthCtx := context.Background()
+	httpClient := &http.Client{Timeout: 2 * time.Second}
+	oauthCtx = context.WithValue(oauthCtx, oauth2.HTTPClient, httpClient)
+
 	client, err := lakekeeper.NewAuthSourceClient(ctx, &core.OAuthTokenSource{
-		TokenSource: oauthConfig.TokenSource(ctx),
+		TokenSource: oauthConfig.TokenSource(oauthCtx),
 	}, c.BaseURL, opts...)
 	if err != nil {
 		return nil, err

